runtime/src/main/java/org/corfudb/runtime/view/SequencerView.java

```diff
@@ -20,26 +21,44 @@ public SequencerView(CorfuRuntime runtime) {
     }
 
     /**
-     * Return the next token in the sequencer for a particular stream.
+     * Return the next token in the sequencer for the global tail or the tails
+     * of multiple streams.
      *
-     * <p>If numTokens == 0, then the streamAddressesMap returned is the last handed out token for
-     * each stream (if streamIDs is not empty). The token returned is the global address as
-     * previously defined, namely, max global address across all the streams.</p>
+     * @param streamIds the streams to query
+     * @return the global tail or a list of tails
+     */
+    public TokenResponse queryTails(UUID ... streamIds) {
+        if (streamIds.length == 0) {
+            return layoutHelper(e -> CFUtils.getUninterruptibly(e.getPrimarySequencerClient()
+                    .nextToken(Collections.emptyList(), 0)));
+        } else {
+            return layoutHelper(e -> CFUtils.getUninterruptibly(e.getPrimarySequencerClient()
```

Comment:
might be a good idea to make the sequencer client take an array instead as well instead of having this extra transform

Reply:
I tried that, but there's a lot of code that uses method from the collection. Changing those code paths to deal with arrays is a bit messy. Also, Arrays.asList is just a wrapper since UUID... streamIds is already an array.